Flat/src/ModulesJets.cc

```diff
@@ -251,19 +258,32 @@ void PandaAnalyzer::JetBasics()
 
 }
 
-// basic jet info for more jtes
+// basic jet info for more jets
 // Responsible: B. Maier
 void PandaAnalyzer::JetHbbBasics(const panda::Jet& jet)
 {
   float csv = (fabs(jet.eta())<2.5) ? jet.csv : -1;
   float cmva = (fabs(jet.eta())<2.5) ? jet.cmva : -1;
   int N = cleanedJets.size()-1;
-  gt->jetPt[N]=jet.pt();
-  gt->jetPtUp[N]=jet.ptCorrUp;
-  gt->jetPtDown[N]=jet.ptCorrDown;
+  if (!isData && analysis->hbb) { 
+    gt->jetPt[N] = jet.ptSmear;
+    gt->jetPtUp[N] = jet.ptSmear*jet.ptCorrUp/jet.pt();
+    gt->jetPtDown[N] = jet.ptSmear*jet.ptCorrDown/jet.pt();
+    gt->jetPtSmearUp[N] = jet.ptSmearUp;
+    gt->jetPtSmearDown[N] = jet.ptSmearDown;
+    TLorentzVector smearedJetP4; 
+    smearedJetP4.SetPtEtaPhiM(jet.ptSmear, jet.eta(), jet.phi(), jet.m());
+    gt->jetE[N]=smearedJetP4.E();
+  } else {
+    gt->jetPt[N] = jet.pt();
+    gt->jetPtUp[N] = jet.ptCorrUp;
+    gt->jetPtDown[N] = jet.ptCorrDown;
+    gt->jetPtSmearUp[N] = jet.pt();
+    gt->jetPtSmearDown[N] = jet.pt();
+    gt->jetE[N]=jet.e();
+  }
```

Comment:
Two things:
- To reduce code duplication, can I suggest you introduce a smear factor that's set to 1 if `isData || !analysis->hbb` and propagate that through? 
- And do you really want to scale pT but not jet mass with the smearing? I'm actually not sure what's right here, but this looks odd to me

Reply:
so this was my intuition after some experience with the b jet regression. if the energy and pt were both scaled up by the same factor, you can end up with a dijet object having negative mass. preserving the mass and only increasing the non-rest energy seemed OK, but I'm not sure what is most correct